src/test/java/org/broadinstitute/hellbender/tools/exome/FilterSamplesIntegrationTest.java

```diff
@@ -0,0 +1,285 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Integration tests for {@link FilterSamples}.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+public class FilterSamplesIntegrationTest extends CommandLineProgramTest {
+
+    @Override
+    public String getTestedClassName() {
+        return FilterSamples.class.getSimpleName();
+    }
+
+    @Test(dataProvider = "targetExtremeMeanCoverageFilterData")
+    public void testSampleExtremeMeanCoverage(final List<SampleCoverageStats> sampleStats,
+                                              final double minMean, final double maxMean)
+        throws IOException
+    {
+        final File samplesFile = createSampleStatsFile(sampleStats);
+        final List<SampleCoverageStats> leftInSamples = sampleStats.stream()
+                .filter(stats -> stats.mean >= minMean && stats.mean <= maxMean)
+                .collect(Collectors.toList());
+        final List<SampleCoverageStats> leftOutSamples = sampleStats.stream()
+                .filter(stats -> stats.mean < minMean || stats.mean > maxMean)
+                .collect(Collectors.toList());
+        final File outputFile = createTempFile("output", ".tab");
+        final File rejectionFile = createTempFile("reject", ".tab");
+        runCommandLine(new String[] {
+                "-" + StandardArgumentDefinitions.INPUT_SHORT_NAME, samplesFile.getPath(),
+                "-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME, outputFile.getPath(),
+                "-" + FilterSamples.MINIMUM_MEAN_COVERAGE_SHORT_NAME, String.valueOf(minMean),
+                "-" + FilterSamples.MAXIMUM_MEAN_COVERAGE_SHORT_NAME, String.valueOf(maxMean),
+                "-" + FilterTargets.REJECT_OUTPUT_FILE_SHORT_NAME, rejectionFile.getPath(),
+        });
+
+        checkLeftInTargets(leftInSamples, outputFile);
```

Comment:
I detect some copy-and-paste from your last PR -- not that there's anything wrong with that, but you need `Targets` --> `Samples` in a few places!